pkg/k8ssandra/util.go

```diff
@@ -29,3 +35,27 @@ func GetDatacenterForDecommission(kc *api.K8ssandraCluster) string {
 
 	return ""
 }
+
+func GetK8ssandraClusterFromCassDc(ctx context.Context, cassdc *cassdcapi.CassandraDatacenter, client client.Client, logger logr.Logger) *api.K8ssandraCluster {
```

Comment:
Issue: this is wrong, and is causing the bug I found in testing.

Reply:
Very good catch!!
We need to use the sanitized names instead for both the cluster and the dc, otherwise we'll end up with either the wrong name or a name that's incompatible with kubernetes.
I'll fix this.

Reply:
See my comments above, I suspect we need to discuss the use of sanitizedName and similar in this PR. I think what you need to use is actually the cluster name from the labels throughout.